source/common/http/conn_manager_impl.cc

```diff
@@ -1438,6 +1438,10 @@ void ConnectionManagerImpl::ActiveStream::refreshDurationTimeout() {
 }
 
 void ConnectionManagerImpl::ActiveStream::refreshCachedRoute(const Router::RouteCallback& cb) {
+  if (cached_route_blocked_) {
```

Comment:
My knowledge on when `refereshCachedRoute()` is called is limited, but if there's an expectation that the function will change the internal state (calling `setRoute()`), then the added code may break this.
I suggest adding a comment explaining the reasoning behind this.

Reply:
get it.